services/bonita-classloader/bonita-classloader-impl/src/main/java/org/bonitasoft/engine/classloader/VirtualClassLoader.java

```diff
@@ -136,4 +162,21 @@ public boolean addListener(ClassLoaderListener listener) {
     public boolean removeListener(ClassLoaderListener classLoaderListener) {
         return listeners.remove(classLoaderListener);
     }
+
+    public void addChild(VirtualClassLoader virtualClassLoader) {
+        children.add(virtualClassLoader);
+    }
+
+    private void removeChild(VirtualClassLoader child) {
+        children.remove(child);
+
+    }
+
+    public boolean hasChildren() {
+        return !children.isEmpty();
+    }
+
+    public Set<VirtualClassLoader> getChildren() {
+        return children;
```

Comment:
May be protected by Collections.unmodifiableSet() to avoid by pass of addChild().